### PR TITLE
Add filter logic

### PR DESCRIPTION
**Pull request by pasankg**

### CODE REVIEW

File: src/components/FilterSection.tsx [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import React from "react";
 import Grid from "@mui/system/Grid";
-import { DateFilter, GenderFilter, AgeRangeFilter } from "./shared";
+import {
+  DateFilter,
+  GenderFilter,
+  AgeRangeFilter,
+  NameMultiSelect,
```
Comment (ranidu) [inline src/components/FilterSection.tsx:7]: should me MultiSelect. because this is a generic component

File: src/components/FilterSection.tsx [hunk 2]
```diff
@@ -9,6 +14,10 @@ const FilterSection: React.FC = () => {
         Data table filters.
       </h3>
       <Grid container spacing={3}>
+        <Grid size={4} spacing={1}>
+          <h3>By Name:</h3>
```
Comment (ranidu) [inline src/components/FilterSection.tsx:18]: replace this h3 tag with typography `/shared`. to maintain consistency of the project

File: src/slices/userFilterSlice.ts
Comment (ranidu) [inline src/slices/userFilterSlice.ts:1]: this logics should be handle inside component level. in here we are breaking Single Responsibility Principle

**Comment by pasankg** [comment]
New PR created

**Pull request closed without merging**